src/NumericalAlgorithms/Interpolation/CubicSpline.cpp

```diff
@@ -0,0 +1,62 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/Interpolation/CubicSpline.hpp"
+
+#include <cstddef>
+#include <gsl/gsl_spline.h>
+#include <pup.h>
+
+#include "ErrorHandling/Assert.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace intrp {
+CubicSpline::CubicSpline(std::vector<double> x_values,
+                         std::vector<double> y_values) noexcept
+    : x_values_(std::move(x_values)), y_values_(std::move(y_values)) {
+  ASSERT(x_values_.size() == y_values_.size(),
+         "The x-value and y-value vectors must be of the same length, but "
+         "received x-value of size: "
+             << x_values_.size()
+             << " and y-value of size: " << y_values_.size());
+  initialize_interpolant();
+}
+
+void CubicSpline::gsl_interp_accel_deleter::operator()(
+    gsl_interp_accel* acc) noexcept {
+  gsl_interp_accel_free(acc);
+}
+
+void CubicSpline::gsl_spline_deleter::operator()(gsl_spline* spline) noexcept {
+  gsl_spline_free(spline);
+}
+
+void CubicSpline::initialize_interpolant() noexcept {
+  size_t num_points = x_values_.size();
```

Comment:
If the `num_points` is not being modified, please make it `const`. If it is, then please add a comment that states what modifies it.